src/uu/ls/src/ls.rs

```diff
@@ -537,19 +539,45 @@ fn extract_time(options: &clap::ArgMatches) -> Time {
     }
 }
 
+/// Parse the dircolors terminal list into a RegexSet and check against
+/// current terminal if applicable
+///
+/// # Returns
+///
+/// Boolean
+fn check_env_variables_against_dircolors(dircolors: &str) -> bool {
+    let ls_colors = env::var("LS_COLORS").unwrap_or(String::new()).is_empty();
```

Comment:
Are you sure that the case where `LS_COLORS` is empty and where it is missing have the same effect?

Reply:
Hi, sorry for taking so long. I just did the test:
When `LS_COLORS` is set and empty or unset -> use TERM or COLORTERM
below only applies when above is true
when `TERM` is empty, unset or not in `DIRCOLORS.h` and `COLORTERM` is empty or unset
do not color even when `--color=always`
when `COLORTERM` has any value or `TERM` has a glob value fitting with the table in `DIRCOLORS.h` use internal default values for coloring